python/pyarrow/tests/test_util.py

```diff
@@ -207,3 +209,19 @@ def test_signal_refcycle():
         assert wr() is not None
         _break_traceback_cycle_from_frame(sys._getframe(0))
         assert wr() is None
+
+
+@pytest.mark.skipif(sys.platform != "win32",
+                    reason="Timezone database is already provided.")
+def test_download_tzdata_on_windows():
+    # Download timezone database
+    download_tzdata_on_windows()
+
+    # Inspect the folder
+    tzdata_path = os.path.expandvars(r"%USERPROFILE%\Downloads\tzdata")
+    tzdata_zones_path = os.path.join(tzdata_path, "windowsZones.xml")
+    assert os.path.exists(tzdata_path)
+    assert os.path.exists(tzdata_zones_path)
+    # 30 files in tzdata (2023) + compressed tzdata.tar.gz + windowsZones.xml
+    assert len(os.listdir(tzdata_path)) == 32
```

Comment:
We might not want to hardcode a number here given that we fetch the latest version, and so this number can then change at any time?

Reply:
Yes, either remove completely, or maybe check the presence of a certain file we are sure will be there (eg `europe`).

You can also verify explicitly the extra `windowsZones.xml` is contained in the directory

Reply:
> You can also verify explicitly the extra windowsZones.xml is contained in the directory

This is done a bit up in the code (`assert os.path.exists(tzdata_zones_path)`). Changed the test so it is a bit clearer:
https://github.com/apache/arrow/pull/38179/commits/55a74797eab41d9139db65bfd3b8ca1f00f91781